flang/lib/Semantics/check-omp-structure.cpp

```diff
@@ -2304,13 +2304,6 @@ void OmpStructureChecker::Leave(const parser::OpenMPFlushConstruct &x) {
   auto &flushList{std::get<std::optional<parser::OmpArgumentList>>(x.v.t)};
 
   if (flushList) {
-    for (const parser::OmpArgument &arg : flushList->v) {
-      if (auto *sym{GetArgumentSymbol(arg)}; sym && !IsVariableListItem(*sym)) {
```

Comment:
@kparzysz is this an issue with the function `IsVariableListItem`?
```
A variable list item is one of the following:
2 • a variable that is not coindexed and that is not a substring;
3 • an array section that is not coindexed and that does not contain an element that is a substring;
4 • a named constant;
5 • an associate name that may appear in a variable definition context; or
6 • a common block name (enclosed in slashes).
```

Reply:
I wonder whether the following is the correct fix here.

```
 bool OmpStructureChecker::IsVariableListItem(const Symbol &sym) {
-  return evaluate::IsVariable(sym) || sym.attrs().test(Attr::POINTER);
+  return evaluate::IsVariable(sym) || sym.attrs().test(Attr::POINTER) ||
+      sym.detailsIf<semantics::CommonBlockDetails>();
 }
 ```

Reply:
My worry was whether that would affect many other constructs. Especially through this use:
```c++
void OmpStructureChecker::Enter(const parser::OmpClause &x)
```

Even if it is wrong that those constructs currently do not accept common blocks, it is unlikely that they will all generate correct code in this case.